src/helm/proxy/retry.py

```diff
@@ -48,7 +48,13 @@ def retry_if_request_failed(result: Union[RequestResult, TokenizationRequestResu
     """Fails if `success` of `RequestResult` or `TokenizationRequestResult` is false."""
     if not result.success:
         hlog(result.error)
-    return not result.success
+    retry_if_fail: bool = (
+        not hasattr(result, "error_flags")
```

Comment:
don't think this is needed?

Reply:
it is because `TokenizationRequestResult` has no error_flages.
At some point I added the error_flags to `TokenizationRequestResult` but since it was not used, I thought it was bad practice.
Additionally, I seem to remember seeing that this function is called with other results than just `RequestResult` and `TokenizationRequestResult` (I think it was `DecodeRequestResult`). In the end, it became a nightmare to add these `error_flags` everywhere, so I thought this was a clean solution